zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -133,71 +135,125 @@ public boolean parseResource(HttpMessage message, Source source, int depth) {
             }
             // Get method and action
             String method = form.getAttributeValue("method");
-            String action = form.getAttributeValue("action");
-            getLogger().debug("Found new form with method: '" + method + "' and action: " + action);
 
-            // If no action, skip the form
-            if (action == null) {
```

Comment:
I removed this check and wrapped the rest into a for loop over the form actions retrieved, the formatting make it scarier than it actually is